### DISCUSSION TITLE
Using PHPMailer

### DISCUSSION THREAD

**Discussion by iniacor** [vercel/community#2844] [Help]
### Summary

I added simple html form and PHPMailer with SMTP , but when i try submit form data, i get 404 error status. What should i do to solve this issue?

### Steps to Reproduce

Just add to form correct data and click submit button("Отправить")

**Comment by smaeda-ks** [comment]
Sounds like you have some issues in your project setup:
https://github.com/vercel-community/php#%EF%B8%8F--usage

But hard to say without having a reproducible example. Could you create one and share that with us?
https://vercel.com/guides/creating-a-minimal-reproducible-example

**Comment by iniacor** [comment]
Thanks for the reply! The problem is that I don't know how to write the vercel.json configuration correctly. I have this problem not only in this project. Please tell me what I need to add to my project or write in vercel.json so that PHPMailer works for me and I don’t get a 405 error. I'll try to find answer from link that you wrote. If nothing helps, I'll ask again

**Comment by iniacor** [comment]
unfortunately I couldn’t figure it out, I added the api folder as well as the vercel.json file, but I still get a 405 error when submitting data from the form. Please help me solve this issue

**Reply by smaeda-ks** [reply]
As I mentioned, I can't investigate this further unless you can provide a minimal reproducible example:
https://vercel.com/guides/creating-a-minimal-reproducible-example

**Comment by iniacor** [comment]
To be honest, I don’t quite understand what exactly I should provide as an example of an error, because I described a case in which an error occurs and what kind of error occurs. After filling in the form fields and clicking on the submit button, a 405 error occurs (405 Method Not Allowed )

**Reply by smaeda-ks** [reply]
@iniacor I need to check your actual code, and that's what I mean by a "minimal reproducible example repo". That information is not enough to answer your questions accurately. Otherwise, this will just be a back-and-forth conversation because everything is just a "guess", and that's not how we solve technical issues.

**Comment by iniacor** [comment]
@smaeda-ks  Uderstood, this is my public git repo with code : https://github.com/iniacor/form-test .Is it will be enougth?

**Comment by iniacor** [comment]
@smaeda-ks  I've just gotten info from form. I see issue was solved.Could You tell, what is the problem was?

**Reply by smaeda-ks** [reply]
see my PR: https://github.com/iniacor/form-test/pull/1

You should have used Composer property rather than including the library manually that way. Other than that, you had some wrong file path in your PHP file, and you could have just moved your static assets into `/public` folder so that no custom routes are required for them

**Reply by smaeda-ks** [reply]
Also, please change your Gmail password as soon as possible. You just exposed that publicly.

**Comment by iniacor** [comment]
@smaeda-ks  Ohh, Thanks a LOT! You are mag and wisard!) Could You give me link to documentation, where i can find more usefull info  about deploy?

**Comment by iniacor** [comment]
@smaeda-ks Hi, Please help me. I don't understand what i am doing wrong and which steps i should do during deploy when i have PHPMailer in my project. Yesterday after your help I started implement the same functionality in another project. But it don't work. Finally i made clone of yesterday form, but it does not work too! That  blowed my brain. This project is clone - https://github.com/iniacor/powercode-test. It includes same files, but i get 404 error. Tell me please where i did go wrong? And What should i do when start another project? Maybe there is some thing that i don't know that i should do before or after deploy?

**Reply by smaeda-ks** [reply]
Just don't set "Root Directory", leave it empty.
![Screenshot 2023-06-15 at 20 52 58](https://github.com/vercel/community/assets/11495867/4c0ae214-56df-4831-9db5-a35aec168093)

**Comment by iniacor** [comment]
And this is project that you've done yesterday https://github.com/iniacor/form-test/tree/mainit has same code, but works.

**Comment by iniacor** [comment]
Thank you very match!!

**Comment by iniacor** [comment]
@smaeda-ks  Hello! Sorry to bother you, I will be very grateful if you help me again. I got a 404 error again. I use gulp to build the project, but because of this, the path to send_mail.php is lost. Tried all path options in this function async function sendData(data) {
  return await fetch('../send_mail.php', {
    method: 'POST',
    body: data
  });
} but nothing helps. this is link to repo  https://github.com/iniacor/powercode and this is the project https://vercel.com/iniacor/powercode
